### PR TITLE
feat: Update the structure of new Coverage engine release note DOCS-623

### PR DESCRIPTION
**Pull request by nicklem**
Updates the new Coverage engine release note to give users more visibility on the updates to different features, which will happen over some months.

### :eyes: Live preview
* [New Coverage engine](https://docs-623-update-the-structure-of-ne--docs-codacy.netlify.app/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks/)

### CODE REVIEW

File: docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md [hunk 1]
```diff
@@ -3,19 +3,33 @@ rss_title: Codacy release notes RSS feed
...
+| Codacy feature             | Transition status | Notes                                                                                                                                        |
+|----------------------------|-------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
+| Codacy app UI              | Planned           | -                                                                                                                                            |
+| Codacy API                 | Planned           | -                                                                                                                                            |
+| Git provider status checks | Beta              | Data from the new Coverage engine is marked \[beta\]<br>and shown alongside data from the current engine. [Read more below](#status-checks). |
+| GitHub coverage summaries  | Planned           | -                                                                                                                                            |
```
Comment (nicklem) [inline docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md:17]: @codacy/alabama As mentioned during the daily, I'm updating this page to give more visibility to the user what's happening with the transition to the new engine. I'll then update this table as needed as the updates go online.

Let me know if I should add more entries here, if this approach is overkill, or any other observations.

cc @machadoit

File: docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md [hunk 2]
```diff
@@ -3,19 +3,33 @@ rss_title: Codacy release notes RSS feed
...
+| Codacy feature             | Transition status | Notes                                                                                                                                        |
+|----------------------------|-------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
+| Codacy app UI              | Planned           | -                                                                                                                                            |
+| Codacy API                 | Planned           | -                                                                                                                                            |
+| Git provider status checks | Beta              | Data from the new Coverage engine is marked \[beta\]<br>and shown alongside data from the current engine. [Read more below](#status-checks). |
```
Comment (claudiacarpinteiro) [inline docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md:16]: Should we order the table by event date? This phase already occurred, so it should be the first line. 🤔

Comment (claudiacarpinteiro) [inline docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md:16]: I believe we should focus on the main occurrence and leave the details (e.g. data marked beta) to the section below.
Also, what do you think of adding the link in the feature column directly? (I left the original, though)

```suggestion
| [Git provider status checks](#status-checks) | Beta              | The new Coverage engine sends coverage data to your Git provider. This beta data is shown alongside the data from the current engine. [Read more below](#status-checks). |
```

File: docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md [hunk 3]
```diff
@@ -3,19 +3,33 @@ rss_title: Codacy release notes RSS feed
...
+| GitHub coverage summaries  | Planned           | -                                                                                                                                            |
+
+## Status checks
```
Comment (claudiacarpinteiro) [inline docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md:19]: My goal was to match the feature name with the one in the table, and make clear that this is the status check from the new engine.
Maybe it's a bit long... Feel free to disregard it. 😉 

```suggestion
## Git provider status checks from the new Coverage engine
```

File: docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md [hunk 4]
```diff
@@ -3,19 +3,33 @@ rss_title: Codacy release notes RSS feed
...
+
+!!! info "The following applies to the repositories for which you set Codacy to send pull request coverage status data to your Git provider (see how on [GitHub](../../repositories-configure/integrations/github-integration.md#status-checks), [GitLab](../../repositories-configure/integrations/gitlab-integration.md#pull-request-status), and [Bitbucket](../../repositories-configure/integrations/bitbucket-integration.md#pull-request-status))."
```
Comment (claudiacarpinteiro) [inline docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md:21]: For consistency.

```suggestion
!!! info "This section applies to the repositories for which you set Codacy to send pull request coverage status data to your Git provider (see how on [GitHub](../../repositories-configure/integrations/github-integration.md#status-checks), [GitLab](../../repositories-configure/integrations/gitlab-integration.md#pull-request-status), and [Bitbucket](../../repositories-configure/integrations/bitbucket-integration.md#pull-request-status))."
```

**Review by claudiacarpinteiro — COMMENTED** [review]
@nicklem This restructuring looks very good! 👍 
I left some comments for your evaluation.

File: docs/release-notes/index.md [hunk 5]
```diff
@@ -18,7 +18,7 @@ For product updates that are in progress or planned [visit the Codacy public roa
 
 2023
 
--   [New Coverage engine status checks November 23, 2023](cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md)
+-   [New Coverage engine November 23, 2023](cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md)
```
Comment (claudiacarpinteiro) [inline docs/release-notes/index.md:21]: Note: It seems that Vale is not very fond of "rollout", though it's a [valid word](https://www.merriam-webster.com/dictionary/rollout)... 🤔 

```suggestion
-   [Rollout of new Coverage engine November 23, 2023](cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md)
```

File: docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md (hunk 1 above)
Comment (machadoit) [reply docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md]: I noticed that we didn't have a doc task on the summary changes https://codacy.atlassian.net/browse/ALA-589 we are now putting it live as time of writing. 

So we can update the docs to 'Live' on the check? With whatever notes you see suitable. Or if you prefer, feel free to add the task on the epic to update on a different PR.

Overall the changes look pretty cool! 👌

**Review by machadoit — APPROVED** [review]
LGTM! (we still doing some QA on it, will let you know)

**Comment by machadoit** [comment]
Actually @nicklem can we add a nudge to review the permissions on GitHub like we added on the UI? 
image

Maybe in general, and then on the notes of status / summary we state that it needs to be updated?

**Review by machadoit — APPROVED** [review]
🚢

**Pull request merged**